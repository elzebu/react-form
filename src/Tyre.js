import React from 'react'
import Brand from './Brand'

import PropTypes from 'prop-types'

const Tyre = (props) => {
    return (
        <div className="tyre border-bottom p-2">
            <h2>test {props.data.name}</h2>
            <Brand data={props.data.brand} />
            <div dangerouslySetInnerHTML={{ __html: props.data.description }}></div>
        </div>
    )
}

Tyre.propTypes = {
    data: PropTypes.shape({
        name: PropTypes.string,
        description: PropTypes.string
    })
}

export default Tyre