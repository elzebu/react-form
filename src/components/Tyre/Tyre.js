import React from 'react'
import PropTypes from 'prop-types'

import Brand from '../Brand/Brand'

const Tyre = (props) => {
    return (
        <div className="tyre border-bottom p-2">
            <h2>{props.data.name}</h2>
            <Brand data={props.data.brand} />
            <div dangerouslySetInnerHTML={{ __html: props.data.description }}></div>
        </div>
    )
}

Tyre.propTypes = {
    data: PropTypes.shape({
        name: PropTypes.string,
        description: PropTypes.string,
        brand:  PropTypes.object
    })
}

export default Tyre