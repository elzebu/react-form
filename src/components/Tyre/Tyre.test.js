import React from 'react'
import { create } from 'react-test-renderer'
import { render } from '@testing-library/react'

import Tyre from './Tyre'


describe('<Tyre />', () => {

    let data
    beforeEach(() => {
        data = {
            name: 'name',
            description: 'description',
            brand: {
                name: 'brandName',
                src: ''
            }
        }
    })
    
    it('matches previous snapshot', () => {
        
        const component = create(
            <Tyre data={data} />,
        )
        let brand = component.toJSON()
        expect(brand).toMatchSnapshot()
    })

    it('displays tyre name in title ', () => {

        const { container } = render(<Tyre data={data} />)

        const title = container.querySelector('h2')

        expect(title.innerHTML).toContain(data.name)
    })

})

