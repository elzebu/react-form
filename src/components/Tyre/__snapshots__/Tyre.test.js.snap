// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tyre /> matches previous snapshot 1`] = `
<div
  className="tyre border-bottom p-2"
>
  <h2>
    name
  </h2>
  <div
    className="brand row"
  >
    <div
      className="col"
    >
      <h3>
        <img
          alt=""
          height="50"
          src=""
        />
         
        brandName
      </h3>
    </div>
    <div
      className="col text-right"
    />
  </div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "description",
      }
    }
  />
</div>
`;
